File: src/app/shared/models/Project.ts
import { Visibility } from "./Visibility";

export interface Project {
    Id?: string;

    Name?: string | null;

    Archived?: boolean;

    ProjectCreatedOn?: string | null;

    Description?: string | null;

    ForksCount?: number;

    HtmLUrl?: string | null;

    ProjectId?: number;

    Language?: string | null;

    Licence?: string | null;

    OwnerName?: string | null;

    OwnerHtmlUrl?: string | null;

    Size?: number;

    Stars?: number;

    WatchersCount?: number;

    ProjectUpdatedOn?: string | null;

    Visibility?: Visibility | null;

    UserId?: string | null;

    CreatedOn?: string | null;

    CreatorId?: string | null;

    UpdatedOn?: string | null;

    UpdaterId?: string | null;
}

File: UpdateData/Models/Project.cs
namespace UpdateData.Models;

public class Project:AuditableEntity
{
    public Guid Id { get; set; }
        
    public long ProjectId { get; set; }

    public string Name { get; set; }
        
    public string Description { get; set; }

    public bool Archived { get; set; }
        
    public int forksCount { get; set; }

    public string HtmLUrl { get; set; }
        
    public string Language { get; set; }
        
    public string Licence { get; set; }
        
    public string OwnerName { get; set; }

    public string OwnerHtmlUrl { get; set; }
        
    public long Size { get; set; }

    public int stars { get; set; }

    public int WatchersCount { get; set; }
    
    public Visibility Visibility { get; set; }

    public DateTime ProjectCreatedOn { get; set; }

    public DateTime ProjectUpdatedOn { get; set; }
        
    public string UserId { get; set; }
}